app/src/main/java/de/westnordost/streetcomplete/util/math/FlatEarthMath.kt

```diff
@@ -0,0 +1,86 @@
+@file:Suppress("NonAsciiCharacters")
+
+package de.westnordost.streetcomplete.util.math
+
+import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
+import de.westnordost.streetcomplete.util.ktx.asSequenceOfPairs
+import kotlin.math.PI
+import kotlin.math.abs
+import kotlin.math.sqrt
+
+/** Calculate stuff assuming an (almost) flat Earth. The main exception from assuming a completely
+ *  flat earth is use of cos(lat) to take into account decreasing longitude distance at
+ *  high latitudes.
+ *  Optimized for performance with precision within 1 m of spherical functions for up to
+ *  0.03° difference between points (several km at common latitudes). */
```

Comment:
Does the math take into account the wrapping at 180th longitude? It should probably be mentioned in the doc comment.

(Also it would be nice if you added the doc comments from their sphericalearthmath siblings to the public functions. Not a merge-blocker.)

Reply:
> Does the math take into account the wrapping at 180th longitude

Not yet, will update and add documentation.

Reply:
> Does the math take into account the wrapping at 180th longitude

Things got a bit too complicated for for `flatDistanceToArc`, so it will fall back to the spherical version in that case.